File: leetcode2681/Solution.py
from typing import List


class Solution:
    def sumOfPower(self, nums: List[int]) -> int:
        n,M = len(nums),int(1e9+7)
        nums.sort()
        minSum = nums[0]
        cur = nums[0]**2*minSum%M
        for i in range(1,n):
            minSum = (minSum*2-nums[i-1]+nums[i])%M
            cur = (cur+nums[i]**2*minSum)%M
        return cur

File: leetcode2681/src/Solution.java
import java.util.Arrays;

class Solution {
    public static final int M = (int) 1e9+7;
    public int sumOfPower(int[] nums) {
        int n = nums.length;
        Arrays.sort(nums);
        long minSum = nums[0];
        long cur = (long) nums[0]*nums[0]%M*minSum%M;
        for (int i = 1; i < n; i++) {
            minSum = (minSum*2-nums[i-1]+nums[i])%M;
            cur = (cur + (long) nums[i]*nums[i]%M*minSum%M)%M;
        }
        return (int) cur;
    }
}